### ISSUE TITLE
Throw on failure instead of exiting

### ISSUE THREAD

**Issue by varunsh-xilinx** [an-tao/trantor#261]
If there are socket errors like if the port [fails to bind ](https://github.com/an-tao/trantor/blob/master/trantor/net/inner/Socket.cc#L68), could Trantor throw an exception rather than exiting? In this case, the calling application could try again with a different port. Or if the exception isn't caught, then it would fail with a visible error and stack trace rather than just a message in the log.

**Comment by cjserio** [comment]
Yes PLEASE! This is shutting down my Drogon application in which the web-server is not a critical component! Let the application decide what to do!
